third-culture-software/bhima
# Improve patient finding functionality

Improve the functionality of finding patients based on name, and optionally gender and data of birth.   This is a general server functionality for patients.   It is implemented initially in the "Lookup Returning Patient" modal in the patient registration page.

**TESTING:**
- Use the IMCK database
- Note that this PR adds a yarn dependency (jaro-winkler).  Be sure to do 'yarn' when testing the PR!
- On the Hospital > Patient Registration page, click on the [Lookup Returning Patient] on the top right
- Click on the "By Name" tab and try some searches.  Once a non-unique search returns, a list of potential names will be given.  If any are the match, click on the name to get the Patient info (for admitting, for instance).
- Here are few search suggestions:
  - BUKUMBA BADIBANGA   (should produce 3 matches, one exact)
  - TUBANGE MUTEBA ARLETTE  (should produce 2 exact matches, eg dupes)
  - TUAKULE TSHIPAMBA (should produce 3 matches with different DOBs)

Note that there are tests in `test/integration/patientMatches.js`
